cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java

```diff
@@ -188,6 +188,18 @@ private void doTestToFromOldId(EntityId id) {
   }
 
   @Test
+  public void testDatasetName() {
+    // TODO: CDAP-5730: Even though dots are allowed, if we use it here, id creation will fail.
```

Comment:
sigh . when we fix this we'll again need upgrade. :(

I think a good thing to do would be to collect all these shortcomings of ``EntityId`` classes and not start using the new ones until we fix all these.